addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4278,32 +4285,43 @@ registry.ImageTools = SnippetOptionWidget.extend({
         new weWidgets.ImageCropWidget(this, this.$target[0]).appendTo(this.options.wysiwyg.$editable);
     },
     transform() {
-        if (this.$target.data('transfo-destroy')) {
-            this.$target.removeData('transfo-destroy');
-            return;
-        }
-        const document = this.$target[0].ownerDocument
+        const document = this.$target[0].ownerDocument;
         this.$target.transfo({ document });
+        const mouseup = this.updateUI.bind(this);
         const mousedown = mousedownEvent => {
             if (!$(mousedownEvent.target).closest('.transfo-container').length) {
                 this.$target.transfo('destroy');
                 $(document).off('mousedown', mousedown);
+                $(document).off('mouseup', mouseup);
             }
-            if ($(mousedownEvent.target).closest('#image-transform').length) {
-                this.$target
-                    .data('transfo-destroy', true)
-                    .attr('style', (this.$target.attr('style') || '')
-                    .replace(/[^;]*transform[\w:]*;?/g, ''));
-            }
-            this.$target.trigger('content_changed');
         };
         $(document).on('mousedown', mousedown);
+        $(document).on('mouseup', mouseup);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this.$target.off('.image_cropped');
+        return this._super(...arguments);
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     */
+    _isTransformed() {
+        return this.$target.is('[style*="transform"]');
+    },
+    /**
+     * @private
+     */
+    _isCropped() {
+        return !!this.$target[0].dataset.aspectRatio;
+    },
```

Comment:
I don't see the need of creating utility method for those if they are called only once.